GiniSwitchSDK/GiniSwitchSDK/Classes/PagesCollectionViewController.swift

```diff
@@ -180,4 +188,60 @@ extension PagesCollectionViewController {
             collectionView.setContentOffset(CGPoint(x: offset, y: 0), animated: true)
         }
     }
+    
+    fileprivate func scrollTo(cell:UICollectionViewCell) {
+        if let collectionView = pagesCollection {
+            let offset = cell.center.x - collectionView.frame.width / 2.0
+            collectionView.setContentOffset(CGPoint(x: offset, y: 0), animated: true)
+        }
+    }
+    
+    fileprivate func snapToClosestCell() {
+        // do everything async because this method is mostly called by collection/scroll view
+        // delegate methods and calling it directly interferes somehow with its innerworkings,
+        // so selecting/deselecting cell might not work properly.
+        DispatchQueue.main.async {
+            guard let collectionView = self.pagesCollection else {
+                return
+            }
+            let cells = collectionView.visibleCells
+            var minDistance = collectionView.contentSize.width
+            let closestCell = cells.reduce(nil, { (closestCell, currentCell) -> UICollectionViewCell? in
+                
```

Comment:
I think this empty line is quite disturbing for reading the anonymous function.

Reply:
It's called a closure, @EarlOfEgo , look it up!
But, yeah, ok